ibm/mas_devops/roles/cp4d_db2wh_backup/tasks/main.yml

```diff
@@ -0,0 +1,126 @@
+---
+# 1. Provide debug information to the user
+# -----------------------------------------------------------------------------
+- name: "DB2 Backup - Debug information"
+  debug:
+    msg:
+      - "CPD Namespace .......................... {{ cpd_meta_namespace_source }}"
+      - "DB2 Instance ID ........................ {{ db2wh_instance_id_source }}"
+      - "DB2 Backup Files Directory ............. {{ db2wh_backup_folder }}"
+
+# 2. Fail if no required keys are provided
+# -----------------------------------------------------------------------------
+- name: "Fail if cpd_meta_namespace_source has not been provided"
+  when: cpd_meta_namespace_source is not defined or cpd_meta_namespace_source == ""
+  fail:
+    msg: "cpd_meta_namespace_source property has not been set"
+
+- name: "Fail if db2wh_instance_id_source has not been provided"
+  when: db2wh_instance_id_source is not defined or db2wh_instance_id_source == ""
+  fail:
+    msg: "db2wh_instance_id_source property has not been set"
+
+- name: "Fail if db2wh_backup_folder has not been provided"
+  when: db2wh_backup_folder is not defined or db2wh_backup_folder == ""
+  fail:
+    msg: "db2wh_backup_folder property has not been set"
+
+# 3. Determine which pod to run in
+# -----------------------------------------------------------------------------
+- name: "Lookup db2 Pod"
+  community.kubernetes.k8s_info:
+    kind: Pod
+    namespace: "{{ cpd_meta_namespace_source }}"
+    label_selectors:
+      - type=engine
+      - app={{ db2wh_instance_id_source }}
+  register: db2wh_pod
+
+- name: Configure facts
+  set_fact:
+    db2wh_pod_name: "{{ db2wh_pod.resources[0].metadata.name }}"
+
+- name: "Debug information"
+  debug:
+    msg:
+      - "Pod name ........... {{ db2wh_pod_name }}"
+  failed_when: db2wh_pod_name is not defined or db2wh_pod_name == ""
+
+# 4. Run backup script in source db2wh pod
+# -----------------------------------------------------------------------------
+- name: "Running backup script on DB2WH pod {{ db2wh_instance_id_source }} as db2inst username... (this can take several minutes)"
```

Comment:
Can we break this up into multiple steps, it will be much easier to debug if each `oc exec` is it's own step instead of all munged together ... e.g. see the changes I made in the db2 hack script where we now have a step to copy the file into the container and a step to execute it: https://github.com/ibm-mas/ansible-devops/blob/master/ibm/mas_devops/roles/cp4d_db2wh_manage_hack/tasks/main.yml#L120-L128

We should also use the oc copy command and a template/local file to get the file into the container (also see the example I provided in the link above) rather than inlining the file in the task itself, makes things easier to read for the user if they for example want to use the role as a guide rather than execute it directly.

Reply:
fixed.